src/theory/quantifiers/extended_rewrite.cpp

```diff
@@ -488,6 +488,8 @@ Node ExtendedRewriter::extendedRewriteIte(Kind itek, Node n, bool full)
     {
       // reverse substitution to opposite child
       // r{ x -> t } = s  implies  ite( x=t ^ C, s, r ) ---> r
+      // We can use ordinary substitute since the result of the substitution
+      // is not being returned.
```

Comment:
Can you say more on why this is not an issue? Why not returning `nn` is OK? It's not used in what is ultimately returned?